core/src/main/java/be/isach/ultracosmetics/cosmetics/deatheffects/DeathEffectFirework.java

```diff
@@ -31,6 +31,6 @@ public void displayParticles() {
         Firework firework = player.getWorld().spawn(player.getLocation(), Firework.class);
         firework.setFireworkMeta(meta);
         firework.setMetadata("uc_firework", new FixedMetadataValue(UltraCosmeticsData.get().getPlugin(), true));
-        Bukkit.getScheduler().runTaskLater(getUltraCosmetics(), firework::detonate, 5);
+        getUltraCosmetics().getScheduler().runAtEntityLater(getPlayer(), firework::detonate, 5);
```

Comment:
Not super familiar with folia so this may be a pointless distinction, but should this not be `runAtEntityLater(firework, ...` since the firework is the entity we're actually interacting with?

Reply:
Location might be more appropriate here, there can be some errors thrown related to scheduling around entities that are despawned. I will change.

Reply:
Need to be careful with scheduling around entities that can despawn, location is more appropriate here on that basis. Refactored!